tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/readonly/TupleVsArrayReadonlyErrorConsistencyTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.types.readonly;
+
+import org.ballerinalang.test.BCompileUtil;
+import org.ballerinalang.test.BRunUtil;
+import org.ballerinalang.test.CompileResult;
+import org.ballerinalang.test.exceptions.BLangTestException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for the readonly violation error consistency between BTuple type and BRecord type.
+ *
+ * @since 2201.10.0
+ */
+public class TupleVsArrayReadonlyErrorConsistencyTest {
```

Comment:
Do we need to mention consistency here? I'd just include readonly field update.

Reply:
The "consistency" keyword was added because the issue being fixed by this PR https://github.com/ballerina-platform/ballerina-lang/issues/42520 is related to an inconsistency between the errors produced during runtime. 

Are you suggesting to modify the test case to match the actual bug? or is it better to use the existing real world test case?

Reply:
I would just have them be part of readonly field tests. If this was the behaviour to start with, we wouldn't have had a consistency issue.

Reply:
Will add them to the place where readonly field tests were defined